### PR TITLE
Cleanup challenge + voting

### PR DESCRIPTION
**Pull request by SUICIDALcow**

### CODE REVIEW

File: src/interfaces/registry.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ interface ApplicantData {
 
 interface Challenge extends RegistryChallenge {
   challengeID: string
```
Comment (robrobbins) [inline src/interfaces/registry.ts]: changed to `challenge` in the other structs (comp, .js)

File: src/redux/action-creators/voting/voting.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+      emitter.unsubscribe()
+    } catch (err) {
+      dispatch(votingVoteError(err))
+    }
+  }
+)
+
+interface CommitVoteParams {
+  challengeID: string
```
Comment (robrobbins) [inline src/redux/action-creators/voting/voting.ts]: challenge

File: src/interfaces/registry.ts (hunk 1 above)
Comment (SUICIDALcow) [reply src/interfaces/registry.ts]: But this is the `Challenge`. Feels weird for `Challenge` to have a `challenge` attribute.

File: src/redux/action-creators/voting/voting.ts (hunk 2 above)
Comment (SUICIDALcow) [reply src/redux/action-creators/voting/voting.ts]: The `Vote` stuff is only partially implemented and not consumed anywhere. I just didn’t want to throw my work away. You alright with us finalizing the param/variable names in the PR for the full implementation?

**Pull request merged**